Source/WebGPU/WebGPU/Device.mm

```diff
@@ -349,7 +355,20 @@ WGPUShaderModule wgpuDeviceCreateShaderModule(WGPUDevice device, const WGPUShade
 
 WGPUSwapChain wgpuDeviceCreateSwapChain(WGPUDevice device, WGPUSurface surface, const WGPUSwapChainDescriptor* descriptor)
 {
-    return WebGPU::releaseToAPI(WebGPU::fromAPI(device).createSwapChain(WebGPU::fromAPI(surface), *descriptor));
+    // FIXME: this should return a swap chain or be removed
+    UNUSED_PARAM(surface);
+    WebGPU::fromAPI(device).createSwapChain(*descriptor);
+    return nullptr;
+}
+
+WGPUTexture wgpuDeviceCreateSwapChainTexture(WGPUDevice device, const WGPUSwapChainDescriptor* descriptor)
+{
+    return WebGPU::releaseToAPI(WebGPU::fromAPI(device).createSwapChain(*descriptor));
+}
+
+WGPU_EXPORT mach_port_t wgpuDeviceGetSwapChain(WGPUDevice device)
```

Comment:
I'm not sure if `mach_port_t` is appropriate to return or if I should return a different type

Reply:
I'm not sure either, but we have something similar already in our WebGL Shared Event extension.

Reply:
Or make a WTF version?

Reply:
Why not do the simple thing of returning a `mach_port_t` instead of smuggling it inside some other object to then have the API user extract it? cf. https://twitter.com/tvaneerd/status/977257752820113413?s=20&t=Er99xQIr9d3Pq-nGAN4r3g

Reply:
I completely agree with doing the simple thing @djg :)

So sounds like leaving it as-is returning the mach_port_t is fine? That would be my preference